### PR TITLE
make an attach file method that does not save

### PR DESCRIPTION
**Pull request by seancookr**
I am trying to use attach file while a before save callback, but the save in attach file begets and infinite loop. Being able to use this method will save that.

### CODE REVIEW

File: lib/dolly/document.rb [hunk 1]
```diff
@@ -116,6 +116,17 @@ def attach_file! file_name, mime_type, body, opts={}
...
+      if opts[:inline]
+        attachment_data = { file_name.to_s => { 'content_type' => mime_type,
+                                                'data'         => Base64.encode64(body)} }
+        doc['_attachments'] ||= {}
+        doc['_attachments'].merge! attachment_data
+      else
+        database.attach id_as_resource, CGI.escape(file_name), body, { 'Content-Type' => mime_type }
+      end
+    end
+
```
Comment (javierg) [inline lib/dolly/document.rb]: please DRY the code... calling on `attach_file!` this method before `save`

Comment (seancookr) [inline lib/dolly/document.rb]: @javierg done

**Comment by javierg** [comment]
:shipit:

**Comment by seancookr** [comment]
Fixed failing tests. anything else?

**Comment by javierg** [comment]
🚢 🇮🇹

**Pull request merged**